app/src/helpers/utils/uploadDocumentValidation.ts

```diff
@@ -0,0 +1,76 @@
+import { UploadDocument, UploadFilesErrors } from '../../types/pages/UploadDocumentsPage/types';
+import { fileUploadErrorMessages } from './fileUploadErrorMessages';
+
+export const uploadDocumentValidation = (uploadDocuments: UploadDocument[]) => {
+    let REGEX_ACCENT_MARKS_IN_NFD = '';
+    for (let i = 300; i < 371; i++) {
+        REGEX_ACCENT_MARKS_IN_NFD += String.fromCharCode(i);
+    }
+    const REGEX_ACCENT_CHARS_IN_NFC = 'À-ž';
+    const REGEX_PATIENT_NAME_PATTERN = `[A-Za-z ${REGEX_ACCENT_CHARS_IN_NFC}${REGEX_ACCENT_MARKS_IN_NFD}]+`;
+    const REGEX_NHS_NUMBER_REGEX = '[0-9]{10}';
+    const REGEX_LLOYD_GEORGE_FILENAME = new RegExp(
+        `[0-9]+of[0-9]+_Lloyd_George_Record_\\[${REGEX_PATIENT_NAME_PATTERN}]_\\[${REGEX_NHS_NUMBER_REGEX}]_\\[\\d\\d-\\d\\d-\\d\\d\\d\\d].pdf`,
+    );
+    const errors: UploadFilesErrors[] = [];
+    const lgFilesNumber = /of[0-9]+/;
+    const FIVEGB = 5 * Math.pow(1024, 3);
+    for (let i = 0; i < uploadDocuments.length; i++) {
```

Comment:
Could you use a for each or for of loop here instead? Also, could the errors.push block be extracted out to a function as it is repeated a few times

Reply:
done